homeassistant/components/somfy/config_flow.py

```diff
@@ -0,0 +1,146 @@
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Instantiate config flow."""
+        self.code = None
+
+    async def async_step_import(self, user_input=None):
+        """Handle external yaml configuration."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='already_setup')
+        return await self.async_step_auth()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        if self.hass.config_entries.async_entries(DOMAIN):
+            return self.async_abort(reason='already_setup')
+
+        return await self.async_step_auth()
+
+    async def async_step_auth(self, user_input=None):
+        """Create an entry for auth."""
+        # Flow has been triggered from Somfy website
+        if user_input:
+            return await self.async_step_code(user_input)
+
+        try:
+            with async_timeout.timeout(10):
+                url, _ = await self._get_authorization_url()
+        except asyncio.TimeoutError:
+            return self.async_abort(reason='authorize_url_timeout')
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected error generating auth url")
+            return self.async_abort(reason='authorize_url_fail')
+
+        return self.async_external_step(
+            step_id='auth',
+            url=url
+        )
+
+    async def _get_authorization_url(self):
+        """Get Somfy authorization url."""
+        from pymfy.api.somfy_api import SomfyApi
+        client_id = self.hass.data[DOMAIN][CLIENT_ID]
+        client_secret = self.hass.data[DOMAIN][CLIENT_SECRET]
+        redirect_uri = '{}{}'.format(
+            self.hass.config.api.base_url, AUTH_CALLBACK_PATH)
+        api = SomfyApi(client_id, client_secret, redirect_uri)
+
+        self.hass.http.register_view(SomfyAuthCallbackView())
+        # Thanks to the state, we can forward the flow id to Somfy that will
+        # add it in the callback.
+        return await self.hass.async_add_executor_job(
+            api.get_authorization_url, self.flow_id)
+
+    async def async_step_code(self, code):
+        """Received code for authentication."""
+        self.code = code
+        return self.async_external_step_done(next_step_id="creation")
+
+    async def async_step_creation(self, user_input=None):
+        """Create Somfy api and entries."""
+        client_id = self.hass.data[DOMAIN][CLIENT_ID]
+        client_secret = self.hass.data[DOMAIN][CLIENT_SECRET]
+        code = self.code
+        from pymfy.api.somfy_api import SomfyApi
+        redirect_uri = '{}{}'.format(
+            self.hass.config.api.base_url, AUTH_CALLBACK_PATH)
+        api = SomfyApi(client_id, client_secret, redirect_uri)
+        token = await self.hass.async_add_executor_job(api.request_token, None,
```

Comment:
It would make sense to validate the code inside `async_step_code`

Reply:
If the code is missing, an error is returned asap by the view: https://github.com/home-assistant/home-assistant/pull/19548/files#diff-05a80de0a3ff7f74840c782130a44750R129
Once the code well received, there is no validation to do.